sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/OffsetSeq.scala

```diff
@@ -180,5 +180,16 @@ object OffsetSeqMetadata extends Logging {
           }
       }
     }
+
+    // SPARK-51187: This incorrect config is not added in the relevantSQLConfs, but the
+    // metadata in the offset log may have this if the batch ran from Spark 3.5.4.
+    // We need to pick the value from the metadata and set it in the new config.
+    // This also leads the further batches to have a correct config in the offset log.
+    metadata.conf.get("spark.databricks.sql.optimizer.pruneFiltersCanPruneStreamingSubplan") match {
```

Comment:
I'll add TODO with mentioning removal when we want to file JIRA ticket for removal immediately.